synapseclient/core/remote_file_storage_wrappers.py

```diff
@@ -46,29 +73,36 @@ def progress_callback(bytes):
 
     @staticmethod
     def download_file(
-        bucket,
-        endpoint_url,
-        remote_file_key,
-        download_file_path,
+        bucket: str,
+        endpoint_url: str,
+        remote_file_key: str,
+        download_file_path: str,
         *,
-        profile_name=None,
-        credentials=None,
-        show_progress=True,
-        transfer_config_kwargs=None,
-    ):
+        profile_name: str = None,
+        credentials: typing.Dict[str, str] = None,
+        show_progress: bool = True,
+        transfer_config_kwargs: dict = None,
+    ) -> str:
         """
         Download a file from s3 using boto3.
 
-        :param bucket:                  name of bucket to upload to
-        :param endpoint_url:            a boto3 compatible endpoint url
-        :param remote_file_key:         object key to upload the file to
-        :param download_file_path:      local path to save the file to
-        :param profile_name:            AWS profile name from local aws config, mutually exclusive with credentials
-        :param credentials:             a dictionary of AWS credentials to use, mutually exclusive with profile_name
-                                        (aws_access_key_id, aws_secret_access_key, aws_session_token)
-        :param show_progress:           whether to print progress indicator to console
-        :param transfer_config_kwargs:  boto S3 transfer configuration (see boto3.s3.transfer.TransferConfig)
-
+        Arguments:
+            bucket: name of bucket to upload to
+            endpoint_url: a boto3 compatible endpoint url
+            remote_file_key: object key to upload the file to
+            download_file_path: local path to save the file to
+            profile_name: AWS profile name from local aws config, mutually exclusive with credentials
+            credentials: a dictionary of AWS credentials to use, mutually exclusive with profile_name
+                                            (aws_access_key_id, aws_secret_access_key, aws_session_token)
```

Comment:
We could format this a little differently. Perhaps something like:
```
            profile_name: AWS profile name from local aws config, mutually exclusive with credentials
            credentials: a dictionary of AWS credentials to use, mutually exclusive with profile_name

                Expected items:

                - `aws_access_key_id`
                - `aws_secret_access_key`
                - `aws_session_token`
            show_progress: whether to print progress indicator to console
```

It would than look like:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/63b924a3-9f40-4c46-b423-61bd4c728f4d)

Thoughts?

Reply:
changed in latest commit